miniscope_io/io.py

```diff
@@ -16,6 +18,78 @@
 from miniscope_io.models.sdcard import SDBufferHeader, SDConfig, SDLayout
 
 
+class BufferedCSVWriter:
+    """
+    Write data to a CSV file in buffered mode.
+
+    Parameters
+    ----------
+    file_path : Union[str, Path]
+        The file path for the CSV file.
+    buffer_size : int, optional
+        The number of rows to buffer before writing to the file (default is 100).
+
+    Attributes
+    ----------
+    file_path : Union[str, Path]
+        The file path for the CSV file.
+    buffer_size : int
+        The number of rows to buffer before writing to the file.
+    buffer : list
+        The buffer for storing rows before writing.
+    """
+
+    def __init__(self, file_path: Union[str, Path], buffer_size: int = 100):
+        self.file_path = file_path
+        self.buffer_size = buffer_size
+        self.buffer = []
+        self.logger = init_logger("BufferedCSVWriter")
+
+        # Ensure the buffer is flushed when the program exits
+        atexit.register(self.flush_buffer)
+
+    def append(self, data: List[Any]) -> None:
+        """
+        Append data (as a list) to the buffer.
+
+        Parameters
+        ----------
+        data : List[Any]
+            The data to be appended.
+        """
+        data = [int(value) if isinstance(value, np.generic) else value for value in data]
+        self.buffer.append(data)
+        if len(self.buffer) >= self.buffer_size:
+            self.flush_buffer()
+
+    def flush_buffer(self) -> None:
```

Comment:
I think this can be blocking because it is not running in the data receive process. I assume timing shouldn't matter much as long as it's not ridiculously slow?